neuralforecast/tsdataset.py

```diff
@@ -370,13 +371,185 @@ def __init__(
+
+            item = dict(
+                temporal=temporal,
+                temporal_cols=temporal_cols,
+                static=static,
+                static_cols=self.static_cols,
+                y_idx=self.y_idx,
+            )
+
+            return item
+        raise ValueError(f"idx must be int, got {type(idx)}")
+
+    def __len__(self):
+        return self.n_groups
+
+    def _as_torch_copy(
+        self,
+        x: Union[np.ndarray, torch.Tensor],
+        dtype: torch.dtype = torch.float32,
+    ) -> torch.Tensor:
+        if isinstance(x, np.ndarray):
+            x = torch.from_numpy(x)
+        return x.to(dtype, copy=False).clone()
+
+    @staticmethod
+    def from_data_directory(
+        files,
+        static_df=None,
+        sort_df=False,
+        id_col="unique_id",
+        time_col="ds",
+        target_col="y",
+    ):
+        """We expect files_ds to have one parquet file per timeseries containing a pd df and static df to be a pd df"""
+
+        # Define indexes if not given
+        if static_df is not None:
+            if isinstance(static_df, pd.DataFrame) and static_df.index.name == id_col:
+                warnings.warn(
+                    "Passing the id as index is deprecated, please provide it as a column instead.",
+                    FutureWarning,
+                )
+            if sort_df:
+                static_df = ufp.sort(static_df, by=id_col)
+
+        # Static features
+        if static_df is not None:
+            static_cols = [col for col in static_df.columns if col != id_col]
+            static = ufp.to_numpy(static_df[static_cols])
+            static_cols = pd.Index(static_cols)
+        else:
+            static = None
+            static_cols = None
+
+        max_size = 0
+        min_size = float("inf")
+        temporal_cols = pd.Index([])
+        last_times = np.array([], dtype=np.datetime64)
+        indices = pd.Series([])
+
+        for file in files:
+            df = pd.read_parquet(file)
+            length = len(df)
+            if length > max_size:
+                max_size = length
+            if length < min_size:
+                min_size = length
+
+            # processor sets y as the first column
+            temporal_cols_temp = pd.Index(
+                [target_col]
+                + [c for c in df.columns if c not in (id_col, time_col, target_col)]
+            )
+            # check all the timeseries have the same temporal cols
+            if len(temporal_cols) == 0:
+                temporal_cols = temporal_cols_temp
+
+            if not temporal_cols_temp.equals(temporal_cols):
+                warnings.warn(
+                    "All the timeseries should have the same temporal columns.",
+                    FutureWarning,
+                )
+
+            ids, times, *_ = ufp.process_df(df, id_col, time_col, target_col)
+            indices = pd.concat([indices, ids])
+            last_times = np.concatenate([last_times, times])
```

Comment:
ids + last_times....these look like they could also be too big to fit in memory? If these are only needed for doing the normalisation, could this be None by default, and in from_dataset have an arg like "process_last_times" or something? that way at least we can keep it optional.

Reply:
there should only be one entry per dataFrame so I think they should probably be ok. As discussed possible future work could involve only having to read the metadata for each parquet file but that's beyond the scope of this PR.